### PR TITLE
Ghosts can eat ghost burgers

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Every tick while a ghost is orbiting a ghost burger there is a 20% chance per ghost (capping at 100% at 5 ghosts) that they will take a bite out of the ghost burger, shrinking it and depleting its reagents until it is completely consumed.
It leaves some ectoplasm behind which you could use to make another burger.
Also ghost burgers can no longer decay or pick up floor germs, the burger moves itself so this would happen reasonably often because it doesn't stay on tables. Also: it is a ghost.

Also just before I posted this PR I noticed that "booberry muffins" also exist so there was nothing for it but to componentise this behaviour and attach it to both food items, so I guess admins can also make anything with reagents edible to ghosts also.

## Why It's Good For The Game

I think the chef should be able to make food for ghosts. It's not fair if only living people get to eat.

## Changelog

:cl:
add: Ghosts (observers) can eat ghost burgers and booberry muffins.
balance: Ghost burgers will not decay or pick up germs due to the fact that they moved themselves off a table.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
Bub buh buh what if they meta game and use the burgers as a y/n ouija board with ghosts to get meta info?????

+1

**Comment by Jacquerel** [comment]
> Bub buh buh what if they meta game and use the burgers as a y/n ouija board with ghosts to get meta info?????

you would have to use muffins for that, ghost burgers don't stay still

File: code/datums/components/food/ghost_edible.dm [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/**
+ * Allows ghosts to eat this by orbiting it
+ * They do this by consuming
```
Comment (MrMelbert) [inline code/datums/components/food/ghost_edible.dm:3]: They do this by consuming

File: code/datums/components/food/ghost_edible.dm [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	initial_reagent_volume = atom_parent.reagents.total_volume
+	notify_ghosts("[parent] is edible by ghosts!", source = parent, action = NOTIFY_ORBIT, header="Something Tasty!")
+
+/datum/component/ghost_edible/RegisterWithParent()
+	. = ..()
+	START_PROCESSING(SSdcs, src)
+
+/datum/component/ghost_edible/UnregisterFromParent()
+	. = ..()
+	STOP_PROCESSING(SSdcs, src)
```
Comment (MrMelbert) [inline code/datums/components/food/ghost_edible.dm:34]:
```suggestion
/datum/component/ghost_edible/RegisterWithParent()
	START_PROCESSING(SSdcs, src)

/datum/component/ghost_edible/UnregisterFromParent()
	STOP_PROCESSING(SSdcs, src)
```

File: code/datums/components/food/ghost_edible.dm (hunk 1 above)
Comment (Jacquerel) [reply code/datums/components/food/ghost_edible.dm:3]: so true

**Comment by Mickyan** [comment]
can you make the ghosts grow bigger when they eat it will be fun

**Comment by Jacquerel** [comment]
> can you make the ghosts grow bigger when they eat it will be fun

I thought about this and also think it would be funny but there isn't really a way to do it which wouldn't look quite bad

**Comment by Wallemations** [comment]
> > can you make the ghosts grow bigger when they eat it will be fun
> 
> 
> 
> I thought about this and also think it would be funny but there isn't really a way to do it which wouldn't look quite bad

Give them a bib so they don't make a mess eating

**Comment by jlsnow301** [comment]
soul (actual spirits)

**Pull request merged**